### PR TITLE
Typography mixins and utility optimization

### PR DESCRIPTION
**Pull request by maulik-thaker1**
## Summary

Optimised  utility and mixins for typography

## ADO Story 

https://dev.azure.com/Kyndryl/Shidoka%20-%20Bridge%20Design%20System/_workitems/edit/1201287

### CODE REVIEW

**Review by peterpham-kyndryl — APPROVED** [review]
Looks good. It'll be better if we can replace some magic numbers with a descriptive var/const. In this loop @for $i from 1 through 4 for example '4' can be assigned to a descriptive var.

File: src/scss/utility/typography.scss [hunk 1]
```diff
@@ -1,63 +1,22 @@
...
-}
-
-.kd-type--headline-06 {
-  @include typography.type-headline-06;
-}
-
-.kd-type--headline-07 {
-  @include typography.type-headline-07;
+// generate display class
+@for $i from 1 through 4 {
```
Comment (brian-patrick-3) [inline src/scss/utility/typography.scss:10]: Display category only has 3 styles.

**Review by brian-patrick-3 — CHANGES_REQUESTED** [review]
1. Any particular reason you didn't put the Body styles in a loop also? Not a big deal just curious.
2. I wonder if you could simplify it even further with a nested loop inside each style for each breakpoint?

**Comment by maulik-thaker1** [comment]
> 1. Any particular reason you didn't put the Body styles in a loop also? Not a big deal just curious.
> 2. I wonder if you could simplify it even further with a nested loop inside each style for each breakpoint/media query above small?

1. Yes. I don't know but body style is used inside root.scss file so while optimizing it, it creates error at compile time.  
  
Screenshot 2023-08-02 at 8 10 22 PM

2. ok will try to keep even shorter if possible.

File: src/scss/utility/typography.scss (hunk 1 above)
Comment (maulik-thaker1) [reply src/scss/utility/typography.scss]: Done. corrected

**Review by brian-patrick-3 — APPROVED** [review]

**Comment by brian-patrick-3** [comment]
> Yes. I don't know but body style is used inside root.scss file so while optimizing it, it creates error at compile time. any reason why we use @include typography.type-body-02; [here](https://github.com/kyndryl-design-system/foundation/blob/beta/src/scss/root.scss#L61) ?

Strange that it errored, but we include that there so the body tag has the default font styles which are inheritable to the rest of the page/components.

I'm going to merge this so I can add a new type style, UI-05, and you can create a new PR if you want to optimize further.

**Comment by maulik-thaker1** [comment]
> > Yes. I don't know but body style is used inside root.scss file so while optimizing it, it creates error at compile time. any reason why we use @include typography.type-body-02; [here](https://github.com/kyndryl-design-system/foundation/blob/beta/src/scss/root.scss#L61) ?
> 
> Strange that it errored, but we include that there so the body tag has the default font styles which are inheritable to the rest of the page/components.
> 
> I'm going to merge this so I can add a new type style, UI-05, and you can create a new PR if you want to optimize further.

sure thanks..!!

**Pull request merged**